api/v1alpha1/tenant_webhook.go

```diff
@@ -95,3 +113,31 @@ func (t *Tenant) ValidateDelete() error {
 	Log.Info("Validating delete for", "tenant", t.Name)
 	return nil
 }
+
+func (t *Tenant) ValidateNodeTypeForXnames(xnames []string, nodeType string, role string) error {
+	ctx, _ := context.WithCancel(context.Background())
+	hsmComponentList, err := GetComponentList(ctx, Log, nodeType, role)
+	if err != nil {
+		return err
+	}
+	var failedXnames []string
+	failedXnames = make([]string, 0, len(xnames))
+
+	for _, xname := range xnames {
+		found := false
+		for _, component := range hsmComponentList.Components {
+			if xname == component.ID {
+				found = true
```

Comment:
you may want to break out of the loop when found

Reply:
great catch -- fixing

Reply:
fixed